### ISSUE TITLE
MD072: expand rule to enforce existence of specified keys

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#662]
### Problem or motivation

Rule MD072 (https://github.com/rvben/rumdl/blob/288f27ce716f28ae2b444068a33c74faec16572b/docs/md072.md) allows one to enforce an order for frontmatter keys, either alphabetical or a specified order.

However, it doesn't catch the case where a key that is configured is missing. Given this example from the documentation (https://github.com/rvben/rumdl/blob/288f27ce716f28ae2b444068a33c74faec16572b/docs/md072.md#custom-key-order):

```toml
[MD072]
enabled = true
key-order = ["title", "date", "author", "tags", "categories"]
```

None of those keys must exist. The rule only checks that if they exist, that they are in the correct order.

### Proposed solution

I think extending the existing rule rather than creating a new rule is better. There might be a new "strict" mode (boolean) where if `key-order` is specified, then all specified keys must exist as well as be in the correct order.

I don't think "strict" mode should stop keys being added to frontmatter that aren't explicitly configured. They should probably work as now where alphabetical order is required, but you can specify any additional keys in frontmatter that you like. "Strict" mode would be just for mandatory keys.

### Alternatives considered

An alternative is a new rule that tests for key existence and leaves order to the existing MD072. A new rule might allow for additional options such as "key must have a value" or "key must be present, but value is optional".

I'm not sure how far `rumdl` should go at validating frontmatter in Markdown files. It can already run existing rules over frontmatter values.

### Feature category

Improve existing rule

**Comment by rvben** [comment]
This shipped in v0.2.13: MD072 now has a `required-keys` option to enforce the presence of frontmatter keys, alongside the existing ordering check. See the MD072 docs for configuration details. Thanks for the suggestion!
